src/components/flyout/deviceReconfigureFlyout.js

```diff
@@ -190,8 +190,10 @@ class DeviceReconfigureFlyout extends React.Component {
     this.state.commonConfiguration.forEach(item => {
       reportedProps[item.label] = item.value;
     });
+    let jobName = this.state.jobInputValue ;
+    jobName = jobName.replace(/[\W_]/g, "");
```

Comment:
Instead of doing the replace in the `apply`, you should do in the onChange for the input fields. That way the user can see in realtime what their job name will be. It's a better user experience with less room for confusion.